src/gh-webhook/github.ts

```diff
@@ -0,0 +1,103 @@
+import crypto from 'node:crypto';
+import type { Env } from '..';
+
+export async function handleGitHubWebhook(request: Request, env: Env): Promise<Response> {
+  if (request.method !== 'POST') {
+    return new Response('Method not allowed', { status: 405, statusText: 'Method Not Allowed' });
+  }
+
+  const githubSecret = process.env.GITHUB_SECRET;
+  const WebhookURL = process.env.WEBHOOK_URL;
+
+  if (!githubSecret || !WebhookURL) {
+    return new Response('Internal server error', { status: 500, statusText: 'Internal Server Error' });
```

Comment:
```suggestion
    return new Response('Internal server error', { status: 400, statusText: 'Missing webhook URL or secret' });
```

Reply:
We can do that but usually I am not a big fan of returning too much information on an endpoint that isn't authenticated yet. This is the bots fault if the variables are incorrect and the 400 ranges usually suggest client error while 500 ranges are for server errors.

Reply:
oh wait i misread the line, yeah a 500 for this is good